health/health.go

```diff
@@ -3,90 +3,134 @@ package health
 // with a valid status
-type ErrInvalidAPIResponse struct {
+type ErrInvalidAppResponse struct {
 	expectedCode int
 	actualCode   int
 	uri          string
 }
 
-// Client represents an api client
+// Client represents an app client
 type Client struct {
-	client rchttp.Clienter
-	name   string
-	url    string
+	Client rchttp.Clienter
+	Check  *health.Check
+	Name   string
+	URL    string
 }
 
-// NewClient creates a new instance of Client with a given api url
-func NewClient(name, url string, maxRetries int) *Client {
+// NewClient creates a new instance of Client with a given app url
+func NewClient(name, url string) *Client {
 	c := &Client{
-		client: rchttp.NewClient(),
-		name:   name,
-		url:    url,
+		Client: rchttp.NewClient(),
+		Name:   name,
+		URL:    url,
+		Check: &health.Check{
+			Name: name,
+		},
 	}
 
-	// Overwrite the default number of max retries on the new client
-	c.client.SetMaxRetries(maxRetries)
+	// healthcheck client should not retry when calling a healthcheck endpoint,
+	// append to current paths as to not change the client setup by service
+	paths := c.Client.GetPathsWithNoRetries()
+	paths = append(paths, "/health", "/healthcheck")
+	c.Client.SetPathsWithNoRetries(paths)
 
 	return c
 }
 
 // Error should be called by the user to print out the stringified version of the error
-func (e ErrInvalidAPIResponse) Error() string {
-	return fmt.Sprintf("invalid response from downstream api - should be: %d, got: %d, path: %s",
+func (e ErrInvalidAppResponse) Error() string {
+	return fmt.Sprintf("invalid response from downstream service - should be: %d, got: %d, path: %s",
 		e.expectedCode,
 		e.actualCode,
 		e.uri,
 	)
 }
 
-// Checker calls an api health endpoint and returns a check object to the caller
-func (c *Client) Checker(ctx *context.Context) (*health.Check, error) {
+// Checker calls an app health endpoint and returns a check object to the caller
+func (c *Client) Checker(ctx context.Context) (*health.Check, error) {
 	logData := log.Data{
-		"api": c.name,
+		"service": c.Name,
 	}
 
-	statusCode, err := c.get(*ctx, "/health")
+	code, err := c.get(ctx, "/health")
 	// Apps may still have /healthcheck endpoint
 	// instead of a /health one
-	if statusCode == http.StatusNotFound {
-		statusCode, err = c.get(*ctx, "/healthcheck")
+	if code == http.StatusNotFound {
+		code, err = c.get(ctx, "/healthcheck")
 	}
 	if err != nil {
-		log.Event(*ctx, "failed to request api health", log.Error(err), logData)
+		log.Event(ctx, "failed to request service health check", log.Error(err), logData)
 	}
 
-	check := getCheck(ctx, c.name, statusCode)
+	currentTime := time.Now().UTC()
+	c.Check.StatusCode = code
+	c.Check.LastChecked = &currentTime
+
+	switch code {
+	case 200:
+		c.Check.Message = StatusMessage[health.StatusOK]
+		c.Check.Status = health.StatusOK
+		c.Check.LastSuccess = &currentTime
+	case 429:
+		c.Check.Message = StatusMessage[health.StatusWarning]
```

Comment:
Do we want the default messages for each code, or shall we use error.Error() ?

Reply:
Simply put yes, I've gone back and forth on this but when I went through the docs with James he referred to the [documentation](https://github.com/ONSdigital/dp/blob/master/standards/HEALTH_CHECK_SPECIFICATION.md#check-statuses) specifically for api clients that only act on status codes

Reply:
See comment below